src/ol/webgl/TileTexture.js

```diff
@@ -278,7 +278,7 @@ class TileTexture extends EventTarget {
       this.textures.push(texture);
 
       const bandCount =
-        textureIndex < textureCount - 1 ? 4 : this.bandCount % 4;
+        textureIndex < textureCount - 1 ? 4 : ((this.bandCount - 1) % 4) + 1;
```

Comment:
I agree this looks like the right fix.  Thanks for identifying the issue and the fix.  Looks like we don't have a test that covers this.  Would you be able to add one?

Reply:
Done.  The problem #13704 identified with 6, 10, 14, etc. bands also affects sources with just 2 bands, so that is presumably a rendering issue related to `LUMINANCE_ALPHA`?